modules/docs/arrow-docs/docs/docs/patterns/typeclasspolymorphism/README.md

```diff
@@ -0,0 +1,711 @@
+
+  @JvmStatic
+  fun main(args: Array<String>): Unit {
+    val user1 = User(UserId("user1"))
+    val user2 = User(UserId("user2"))
+    val user3 = User(UserId("unknown user"))
+
+    val deferredModule = Module(DeferredK.async())
+    deferredModule.run {
+      runBlocking {
+        try {
+          println(repository.allTasksByUser(user1).fix().deferred.await())
+          println(repository.allTasksByUser(user2).fix().deferred.await())
+          println(repository.allTasksByUser(user3).fix().deferred.await())
+        } catch (e: UserNotInRemoteStorage) {
+          println(e)
+        }
+      }
+    }
+  }
+}
+```
+
+As you know, you're in charge to catch exceptions on coroutines. As you can see, implementation details as this catch, 
+or the observable subscriptions from previous examples, are implementation details related to the given data types being 
+used for each case, so those must live here, in the program's edge.
+
+Same result one more time:
+
+```
+[Task(value=LocalTask assigned to user1)]
+[Task(value=Remote Task assigned to user2)]
+UserNotInRemoteStorage(user=User(userId=UserId(value=unknown user)))
+```
+
+There's also an alternative api provided by Arrow a bit more fancier for [`DeferredK`]({{ '/docs/integrations/kotlinxcoroutines/' | relative_url }}). It takes care of runBlocking and awaiting on the operations for you:
+
+```kotlin
+object test {
+
+  @JvmStatic
+  fun main(args: Array<String>): Unit {
+    val user1 = User(UserId("user1"))
+    val user2 = User(UserId("user2"))
+    val user3 = User(UserId("unknown user"))
+
+    val deferredModuleAlt = Module(DeferredK.async())
+    deferredModuleAlt.run {
+          println(repository.allTasksByUser(user1).fix().unsafeAttemptSync())
+          println(repository.allTasksByUser(user2).fix().unsafeAttemptSync())
+          println(repository.allTasksByUser(user3).fix().unsafeAttemptSync())
+        }
+  }
+}
+```
+
+This one wraps the result into [`Try`]({{ '/docs/datatypes/try' | relative_url }}) (which can be `Success` or `Failure`).
+
+```
+Success(value=[Task(value=LocalTask assigned to user1)])
+Success(value=[Task(value=Remote Task assigned to user2)])
+Failure(exception=UserNotInRemoteStorage(user=User(userId=UserId(value=unknown user))))
+```
+
+Finally, let's use a more fancy data type related to Functional Programming: [`IO`]({{ '/docs/effects/io' | relative_url }}). 
+`IO` exists to wrap an in/out operation that causes side effects and make it pure.
+
+```kotlin
+object test {
+
+  @JvmStatic
+  fun main(args: Array<String>): Unit {
+    val user1 = User(UserId("user1"))
+    val user2 = User(UserId("user2"))
+    val user3 = User(UserId("unknown user"))
+
+    val ioModule = Module(IO.async())
+    ioModule.run {
+      println(repository.allTasksByUser(user1).fix().attempt().unsafeRunSync())
+      println(repository.allTasksByUser(user2).fix().attempt().unsafeRunSync())
+      println(repository.allTasksByUser(user3).fix().attempt().unsafeRunSync())
```

Comment:
`unsafeRunSync` throws exceptions for failures, maybe use `unsafeAttemptSync` instead.

Reply:
Syntax non existent (yet?). I think there's not other choice than `attempt().unsafeRunSync()` for now.